### ISSUE TITLE
module 'types' has no attribute ClassType

### ISSUE THREAD

**Issue by tsktsktsk123** [dask/distributed#3636]
Hello,

Enormous fan of dask for a while now. I've just updated to the 2.12 and ran into an issue initializing an object in the a function I pass on to the scheduler. This raises the following error I've been able to replicate on 2.12, and was resolved by downgrading to 2.11.

The code to get the error:

```python
class A:
    def __init__(self):
        self.a = 100
        
def func(x):
    A()
    return x**2

results = client.map(func, range(100))
r = client.gather(results)
```

Perhaps a module within dask is named types.py and thus conflicts? Because if i run getattr(types, 'ClassType') it just evaluates finely within the local context.
 
I hope I've been of help.
Thanks for your great work.

*EDIT*: After resetting the environment I ran into the error once more but this time with a clearer route to cloudpickle. I found a similar issue in the repo if mlflow which is unsolved as of now. I downgraded cloudpickle to 1.2.2 and my code worked again, if when upgrading distributes to 2.12. So my conclusion for now is that a function passed to a worker cannot initialize an object when ran with cloudpickle 1.3.0.
Perhaps exclude cloudpickle==1.3.0 from the current requirements for distributed?

https://github.com/mlflow/mlflow/issues/2429 

My versions to reproduce are:

MacOS
Python 3.7

cloudpickle==1.3.0
distributed==2.11.0 (error was also present in 2.12)
dask==2.12.0

**Comment by TomAugspurger** [comment]
Thanks for the report. To confirm, the snippet you posted above consistently raises an error? I don't see one with cloudpickle 1.3.0 and distributed master. Can you post the traceback?

**Comment by tsktsktsk123** [comment]
I'm not so sure, I'm unable to reproduce the error as of now, while keeping the things same. It's working for me again and I've tried to upgrade the cloudpickle and distributed versions back to their master again but it still works strangely. If I run into it I'll make sure to post the entire traceback.

**Comment by ogrisel** [comment]
cloudpickle does not guarantee to be able to load pickle files from prior versions. Maybe this is a problem with dask workers reusing old results cached on disk? I thought this would not be the case after a cluster restart.

**Comment by ogrisel** [comment]
FYI we released cloudpickle 1.4.1 that with a backward compat fix. However this is weird that a dask cluster would try to load objects pickled with a different Python runtime configuration. It's not guaranteed to work in cloudpickle.
